### ISSUE TITLE
Discrepancy between apply_attention_guided_fusion and Attention Guided Fusion as explained in the paper

### ISSUE THREAD

**Issue by Christophernph** [zjx0101/ObjectClear#23]
First of all, thank you for your work and making it available to the community.

I was having a look at the implementation of Attention Guided Fusion in the code, specifically, what happens when enabling the flag `apply_attention_guided_fusion` [here](https://github.com/zjx0101/ObjectClear/blob/bc8743eb6639d0eda67b3f1d1b68286f654d15f3/objectclear/pipelines/pipeline_objectclear.py#L1917C32-L1917C61).  From what I can tell, the actual cross-attention activations are only used in the last step (when `i == len(timesteps) - 1`), and even then, only for outputting the attention map as images, not for blending. Is this a mistake? I would have assumed something like what is being done for `i == 0` should be applied throughout the diffusion process, but instead of using the original mask, the cross-attention activations should be used, i.e. something like:

```python
attn_key, attn_map = next(iter(self.cross_attention_scores.items()))
attn_map = self.resize_attn_map_divide2(attn_map, mask, fuse_index) # Resize to mask shape and map to [0,1]
init_latents_proper = image_latents # Original image latents (no noise)
if self.do_classifier_free_guidance:
     _, init_mask = attn_map.chunk(2)
else:
     init_mask = attn_map

noise_timestep = timesteps[i + 1]
init_latents_proper = self.scheduler.add_noise(
         init_latents_proper, noise, torch.tensor([noise_timestep])
)
latents = (1 - init_mask) * init_latents_proper + init_mask * latents
```

Any insights into this would be appreciated.

**Comment by Christophernph** [comment]
Nevermind, I just saw that you use the final attention map [here](https://github.com/zjx0101/ObjectClear/blob/bc8743eb6639d0eda67b3f1d1b68286f654d15f3/objectclear/pipelines/pipeline_objectclear.py#L2044) after the VAE reconstruction. It would also be interesting to explore progressively blending/locking the original image throughout the diffusion process. Anyways, I'm closing this issue.
